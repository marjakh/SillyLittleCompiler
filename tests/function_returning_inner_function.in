function foo() {
  function innerhelper() { write(600); }
  function inner() { write(500); innerhelper(); }
  return inner;
}

let f = foo();
f();

function bar() {
  function innerhelper() { write(700); }
  function inner(x) { write(x); innerhelper(); }
  return inner;
}

let b = bar();
b(800);