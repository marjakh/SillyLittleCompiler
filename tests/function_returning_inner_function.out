500
600
800
700
